junit-jupiter-params/src/main/java/org/junit/jupiter/params/converter/DefaultArgumentConverter.java

```diff
@@ -84,6 +91,10 @@ public final Object convert(Object source, Class<?> targetType, ClassLoader clas
 		}
 
 		if (source instanceof String) {
+			if (targetType == Locale.class && getLocaleConversionFormat() == LocaleConversionFormat.BCP_47) {
+				return Locale.forLanguageTag((String) source);
+			}
```

Comment:
I'm delaying the format retrieval via `getLocaleConversionFormat()` because I'm unsure if it would be safe to grab that value in the constructor and keep it internally.

If you think it could be fetched earlier, I'm happy to refactor it.